dbt/adapters/bigquery/impl.py

```diff
@@ -201,6 +206,15 @@ def list_relations(cls, profile, project_cfg, schema, model_name=None):
             #       won't need to do this
             max_results=100000)
 
+        # This will 404 if the dataset does not exist. It's lazily
+        # evaluated, so wrap it in a list to throw _now_. This behavior
+        # mirrors the implementation of list_relations for other adapters
+
+        try:
+            all_tables = list(all_tables)
```

Comment:
Have you considered consuming the iterator here by moving the list comprehension up? As it is this is going to make two copies of each table (one BQ API result and one Relation). If we're actually pulling up to 100k entries, doubling that could be a lot!

Reply:
I think 100k would be _super_ excessive! But agree, happy to slide the comprehension up into the try/catch